File: 002/soln.py
# Solution to Project Euler problem by Joseph Jude
# 
# Webpage: http://tech.jjude.com/euler
# Git Repo: https://github.com/jjude/euler 
# 
# Problem 002: Sum of even valued Fibonacci numbers less than 4 million
# WIL: tuple assignment

answer = 0
current_fibonacci = 1
next_fibonacci = 2

while current_fibonacci <= 4000000:
  if current_fibonacci % 2 == 0:
    answer += current_fibonacci
  # generate the next numbers
  current_fibonacci, next_fibonacci = next_fibonacci, current_fibonacci + next_fibonacci

print answer
# 4613732

File: 004/soln.swift
/*
Solution to Project Euler problems by Joseph Jude

Webpage: http://tech.jjude.com/euler
Git Repo: https://github.com/jjude/euler

execute with: swift soln.swift
compile with: swiftc soln.swift

Problem 004: Find the largest palindrome made from the product of two 3-digit numbers.
WIL: 
  reverse for loop (there is .reverse() and stride())
  double for loop in swift is slow, pathetically slow
*/

var answer = 0
var k = 0
var s = ""

var i = 100
var j = 100

while i < 1000 {
  while j < 1000 {
    k = i * j
    s = String(k)
    if s == String(s.characters.reverse()) && k > answer {
      answer = k
    }
    j += 1
  }
  i += 1
  j = 100
}

print(answer)
//906609